src/decay.hpp

```diff
@@ -0,0 +1,137 @@
+// ====================================================================
+// This file is part of FlexibleSUSY.
+//
+// FlexibleSUSY is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// FlexibleSUSY is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with FlexibleSUSY.  If not, see
+// <http://www.gnu.org/licenses/>.
+// ====================================================================
+
+#ifndef DECAY_H
+#define DECAY_H
+
+#include <initializer_list>
+#include <unordered_map>
+#include <vector>
+#include <string>
+
+#include <boost/core/demangle.hpp>
+
+namespace flexiblesusy {
+
+class Decay {
+public:
+   Decay(int, std::initializer_list<int>, double, std::string const&);
+   ~Decay() = default;
+   Decay(const Decay&) = default;
+   Decay(Decay&&) = default;
+   Decay& operator=(const Decay&) = default;
+   Decay& operator=(Decay&&) = default;
+
+   int get_initial_particle_id() const { return pid_in; }
+   const std::vector<int>& get_final_state_particle_ids() const {
+      return pids_out;
+   }
+   std::size_t get_final_state_size() const { return pids_out.size(); }
+
+   double get_width() const { return width; }
+   std::string get_proc_string() const { return proc_string; }
+   void set_width(double w) { width = w; }
+
+private:
+   int pid_in{0};
+   std::vector<int> pids_out{};
+   double width{0.};
+   std::string proc_string;
```

Comment:
I believe this could be a `const char* const`, instead of a `std::string`, because the decays to compute is known at meta code time. In this way the size of the `Decay` objects would be significantly reduced. (And we have many decays.)

However, if a `const char* const` is used, we are restricted to string literals `"hh -> {hh, hh}"` etc., i.e. no run-time strings. But I believe this is no issue, because the decays to compute are known at meta code time. (Such a change would also nicely go hand-in-hand with my other comment on removing `wrong_width_sign_msg`.)